src/org/openlcb/hub/Hub.java

```diff
@@ -80,18 +79,29 @@ private void createServerThread() {
         Thread t = new Thread("openlcb-hub-output") {
             @Override
             public void run() {
-                while (!disposed) {
+                while (!disposed) { // loop until termination is requested
                     try {
                         // as items arrive in queue, forward to every available connection
                         Memo m = queue.take();
-                        for ( Forwarding e : threads) {
+                        
+                        ArrayList<Forwarding> threadList;
+                        synchronized (threads) {
+                            threadList = new ArrayList<>(threads); // avoid CME
+                        }
+                        
+                        for ( Forwarding e : threadList) {
```

Comment:
Since we are now iterating beyond the lifetime of the connection, we should make a change to ReaderThread.
in line 231, just before `threads.remove(this);` we should set `output = null` to ensure that the thread object ignores all further forwarding attemtps.

Reply:
Done.  Thanks.